pkg/collector/horizontalpodautoscaler.go

```diff
@@ -135,6 +149,20 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 			metrics = append(metrics, targetCPUUtilization)
 		}
 
+		var minReplicas *int32
+		if otelcol.Spec.Autoscaler.MinReplicas != nil {
+			minReplicas = otelcol.Spec.Autoscaler.MinReplicas
+		} else {
+			minReplicas = otelcol.Spec.MinReplicas
+		}
```

Comment:
nit, but can we avoid those else branches?
```suggestion
		minReplicas := otelcol.Spec.MinReplicas
		if otelcol.Spec.Autoscaler.MinReplicas != nil {
			minReplicas = otelcol.Spec.Autoscaler.MinReplicas
		}
```